cmd/zpool/zpool_main.c

```diff
@@ -9547,16 +9556,46 @@ errors_nvlist(zpool_handle_t *zhp, status_cbdata_t *cb, nvlist_t *item)
 					    ZPOOL_ERR_DATASET, &dsobj) == 0);
 					verify(nvlist_lookup_uint64(nv,
 					    ZPOOL_ERR_OBJECT, &obj) == 0);
-					errl[i] = safe_malloc(len);
-					zpool_obj_to_path(zhp, dsobj, obj,
-					    errl[i++], len);
+					if (cb->cb_verbosity < 2) {
+						errl[i] = safe_malloc(len);
+						zpool_obj_to_path(zhp, dsobj,
+						    obj, errl[i++], len);
+					} else {
+						uint64_t lvl, blkid;
+
+						errnvl[i] = fnvlist_alloc();
+						lvl = fnvlist_lookup_uint64(nv,
+						    ZPOOL_ERR_LEVEL);
+						blkid = fnvlist_lookup_uint64(
+						    nv, ZPOOL_ERR_BLKID);
+						zpool_obj_to_path(zhp, dsobj,
+						    obj, pathbuf, len);
+						fnvlist_add_string(errnvl[i],
+						    "path", pathbuf);
+						fnvlist_add_uint64(errnvl[i],
+						    "level", lvl);
+						fnvlist_add_uint64(errnvl[i++],
+						    "record", blkid);
+					}
```

Comment:
Couldn't we always do it the verbose way here to simplify the code, and just not print the additional information later?  It does not look like very performance-sensitive code path.

Reply:
> enormous nvlists if a single file has many corrupt records.

Along these same lines I'm a bit concerned with logging a line per block.  That could be overwhelming.

Thing about this from a user perspective, I really don't care about how ZFS decided to internally layout the file (objid, level, blkid).  What is useful to me are the file offsets which are corrupt.  That's a little more work to generate but shouldn't be too bad.  Maybe something like:

```
errors: Permanent errors have been detected in the following files:

        /testpool/randfile.7 393216-524288,1048576-1179647,1966080-2097151
        /testpool/randfile.5 917504-1048575
        ...
```